source/editableText.py

```diff
@@ -268,13 +274,10 @@ def script_caret_changeSelection(self,gesture):
 		gesture.send()
 		if isScriptWaiting() or eventHandler.isPendingEvents("gainFocus"):
 			return
-		api.processPendingEvents(processEventQueue=False)
 		try:
-			newInfo=self.makeTextInfo(textInfos.POSITION_SELECTION)
+			waitForAndSpeakSelectionChange(oldInfo)

```

Comment:
I think you want `self.waitForAndSpeakSelectionChange`; note the "self". This line will actually throw an `AttributeError`, but the exception is being caught by the catch-all except clause.